### PR TITLE
Convert taskProperties.gradle from Groovy to Kotlin script

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

There wasn't too much that wouldn't translate 1 to 1 from Groovy to Kotlin. Let me know if there's a more idomatic way to define the logic.

### CODE REVIEW

File: gradle/codeQuality.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ apply plugin: 'codenarc'
 def validateTaskPropertiesForConfiguration(Configuration configuration) {
     // Apply to all projects depending on :core
     configuration.dependencies.withType(ProjectDependency).matching { it.dependencyProject == project(":core") }.all {
-        project.apply from: "$rootDir/gradle/taskProperties.gradle"
+        project.apply from: "$rootDir/gradle/taskProperties.kts"
```
Comment (eskatos) [inline gradle/codeQuality.gradle]: :x: should be `taskProperties.gradle.kts`

File: subprojects/core/core.gradle [hunk 2]
```diff
@@ -16,7 +16,7 @@
 plugins {
     id 'java-library'
 }
-apply from: "$rootDir/gradle/taskProperties.gradle"
+apply from: "$rootDir/gradle/taskProperties.kts"
```
Comment (eskatos) [inline subprojects/core/core.gradle]: :x: should be `taskProperties.gradle.kts`

File: gradle/taskProperties.kts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+gradle.projectsEvaluated {
```
Comment (eskatos) [inline gradle/taskProperties.kts]: :x: This is Groovy :)
Because the filename does not match `.gradle.kts` Gradle falls back to the Groovy DSL.

Comment (bmuschko) [reply gradle/taskProperties.kts]: Doh! User mistake. I feel like we should document that in some form. I'd personally prefer if we'd fail the build if the file extension is not something we recognize. Probably the same thing would happen if you have a typo.

**Comment by bmuschko** [comment]
Closed as duplicate of https://github.com/gradle/gradle/pull/4478 which also turns the custom task into Kotlin.

**Pull request closed without merging**